### PR TITLE
Create new Pangea token type

### PR DESCRIPTION
**Pull request by MichaelCombs28**
Accept Pangea token type in service constructors. Type must be 'pangea_token' or a raw string.

### CODE REVIEW

**Comment by kenany** [comment]
Add a release note to `packages/pangea-node-sdk/CHANGELOG.md` too.

File: packages/pangea-node-sdk/src/services/base.ts [hunk 1]
```diff
@@ -20,15 +20,39 @@ class BaseService {
...
+    token: PangeaToken,
     config: PangeaConfig,
     configID?: string
   ) {
     if (!serviceName) throw new Error("A serviceName is required");
     if (!token) throw new Error("A token is required");
 
     this.serviceName = serviceName;
-    this.token = token;
+    if (typeof token === "string") {
```
Comment (pangea-andrest) [inline packages/pangea-node-sdk/src/services/base.ts:31]: Wouldn't be better to move this whole logic into a function?

**Review by pangea-andrest — APPROVED** [review]
Just one minor comment but approved anyway

File: packages/pangea-node-sdk/src/services/base.ts (hunk 1 above)
Comment (MichaelCombs28) [reply packages/pangea-node-sdk/src/services/base.ts:31]: It's only used here, we can break it out later if more use comes. I'm not a fan of early abstraction, generally abstract only a rule of 3.

**Pull request merged**